src/Illuminate/Database/Console/Migrations/MigrateCommand.php

```diff
@@ -134,26 +136,88 @@ protected function prepareDatabase()
     protected function repositoryExists()
     {
         return retry(2, fn () => $this->migrator->repositoryExists(), 0, function ($e) {
-            if (! $e->getPrevious() instanceof SQLiteDatabaseDoesNotExistException) {
-                return false;
-            }
+            try {
+                if ($e->getPrevious() instanceof SQLiteDatabaseDoesNotExistException) {
+                    return $this->createMissingSqliteDatbase($e->getPrevious()->path);
+                }
 
-            if ($this->option('force')) {
-                return touch($e->getPrevious()->path);
-            }
+                $connection = $this->migrator->resolveConnection($this->option('database'));
 
-            if ($this->option('no-interaction')) {
+                if (
+                    $e->getPrevious() instanceof PDOException &&
+                    $e->getPrevious()->getCode() === 1049 &&
+                    $connection->getDriverName() === 'mysql') {
```

Comment:
Why don't you create a `MySQLDatabaseDoesNotExistException`, instead of this `if` with three conditions (like `SQLiteDatabaseDoesNotExistException` on [L.140](https://github.com/laravel/framework/pull/44153/files#diff-28caf967243e756318d62361df72ec340f0b96c96f439c0bc800360733fee304R140))?